src/Accelerators/CMakeLists.txt

```diff
@@ -12,17 +12,14 @@ endif (ACCELERATORS_TO_BUILD)
 
 # Generate the Accelerators.inc file.
 set(accelerators_inc "${CMAKE_CURRENT_BINARY_DIR}/Accelerators.inc")
+set(create_accelerators_inc_script "${CMAKE_CURRENT_SOURCE_DIR}/CreateAcceleratorsInc.cmake")
 list(APPEND accelerators_inc_generate_commands
-  COMMAND ${CMAKE_COMMAND} -E echo "'#'define APPLY_TO_ACCELERATORS'('MACRO')' \\ " > ${accelerators_inc})
-
-if (ACCELERATORS_TO_BUILD) 
-  foreach(t ${ACCELERATORS_TO_BUILD})
-    list(APPEND accelerators_inc_generate_commands
-      COMMAND ${CMAKE_COMMAND} -E echo_append "  MACRO'('${t}')' \\ " >> ${accelerators_inc})
-  endforeach(t)
-  list(APPEND accelerators_inc_generate_commands
-      COMMAND ${CMAKE_COMMAND} -E echo >> ${accelerators_inc})
-endif (ACCELERATORS_TO_BUILD)
+  COMMAND ${CMAKE_COMMAND} "-DINC_FILE=${accelerators_inc}.tmp" "-DACCELERATORS=${ACCELERATORS_TO_BUILD}" 
+    -P "${create_accelerators_inc_script}")
+list(APPEND accelerators_inc_generate_commands
```

Comment:
These can be inside the `.cmake` file. For the versions they are not because generation differs based on how we add the llvm version, but here they can all be together and you could pass the name of the `.inc` file you want to create and let the `.cmake` file decide how to create the `.tmp` name and do the copy, etc.